include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -6624,8 +6646,8 @@ FEEvaluationAccess<dim, dim, Number, is_face, VectorizedArrayType>::
 {
   Assert(this->cell != numbers::invalid_unsigned_int, ExcNotInitialized());
   AssertIndexRange(q_point, this->n_quadrature_points);
-  Assert(this->J_value != nullptr, ExcNotInitialized());
-  Assert(this->jacobian != nullptr, ExcNotInitialized());
+  Assert(this->J_value != nullptr, internal::MatrixFreeNotInitialized("???"));
+  Assert(this->jacobian != nullptr, internal::MatrixFreeNotInitialized("???"));
```

Comment:
`update_gradients` (the former would be fine with `update_values` as well, but let's stick to the relevant one here).